node-api/backend/validator.go

```diff
@@ -200,11 +204,21 @@ func (b Backend) ValidatorByID(
 		return nil, errors.Wrapf(err, "failed to get state from slot %d", slot)
 	}
 	index, err := utils.ValidatorIndexByID(st, id)
-	if err != nil {
+	switch {
+	case err == nil:
+		// continue processing
+	case errors.Is(err, collections.ErrNotFound):
+		return nil, ErrValidatorNotFound
+	default:
 		return nil, errors.Wrapf(err, "failed to get validator index by id %s", id)
 	}
 	validator, err := st.ValidatorByIndex(index)
-	if err != nil {
+	switch {
+	case err == nil:
+		// continue processing
+	case errors.Is(err, collections.ErrNotFound):
+		return nil, ErrValidatorNotFound
```

Comment:
I think you don't need to special case here?
Index may not exist but if it exists the validator must exist.
So here you can just
```
if err != nil {
return err
}
```

Reply:
We want to show the specific message `Validator not found`, so done like this. Another way is like you said, not returning the defined error of `ErrValidatorNotFound` and directly returning err and in caller `GetStateValidator ` using `collections.ErrNotFound` in switch case.
Wanted to use our defined error in handler hence doing like this. Let me know your thoughts.

Reply:
Oh, I see, I misunderstood your intent. All good